### ISSUE TITLE
Documentation not relating to the feature which got me here

### ISSUE THREAD

**Issue by hgunter** [MicrosoftDocs/visualstudio-docs#2947]
I was on the Visual Studio Add Item dialogue box, facing a list of Code items available, and wanted to know what those items provide and how they differ, specifically the difference between a Code File and a Module.  The document offers a list of command line interface commands such as "Import and Export Settings Command".  None of the options relates to the five items in the dialogue box (Class, Module, Interface, Code File, COM Class).

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 8d4adf8f-cb53-424e-6dac-85290cf1cc97
* Version Independent ID: fcc518e4-9576-4891-22c1-7618740b80da
* Content: [Add New Item Command - Visual Studio](https://docs.microsoft.com/en-us/visualstudio/ide/reference/add-new-item-command?view=vs-2017#feedback)
* Content Source: [docs/ide/reference/add-new-item-command.md](https://github.com/MicrosoftDocs/visualstudio-docs/blob/master/docs/ide/reference/add-new-item-command.md)
* Product: **visual-studio-windows**
* GitHub Login: @gewarren
* Microsoft Alias: **gewarren**

**Comment by gewarren** [comment]
@hgunter Agreed, the help link should not go to that page. We don't actually have a docs page on the differences between items in the Add New Item dialog box.

It sounds like you're using Visual Basic, correct? I tried to add one of each, and the only difference, other than the default name for the file, is that a module file contains `Module` and `End Module` keywords.

**Comment by hgunter** [comment]
Hi, Genevieve – thanks for the response.

Yes, VB – C# has one typing lots of { } and type conversions and voids and generally drowning in syntax, whereas VB keeps a clear layout and, very importantly, is not case-sensitive (one simply can’t define an Integer named integer, or a Dictionary object named dictionary, and one makes a meaningful name which adds clarity).

I decided to try adding one of each to see what they do, and discovered the result you described.  This is fine for these simpler things, but not good for more complex situations.  A major shortcoming in documentation these days is lack of a description of what a thing is good for.  One gets endless detail of what it breaks down to, but very seldom anything on what it fits into.  One is rather in the position of a manual for a piece of hardware which tells you what ports and internal components it has, but not what to connect to them, so that you have no idea what your computer is for how to make it useful by connecting a keyboard and monitor.

I keep explaining this to people in the hope that they will be in a position, as I am not, to start a shift to more useful documentation formats - as well as what sub-parts it has, what it’s useful for.  I am still in this position with respect to modules and code files: I know what is different about them, but not what purpose each is intended to serve (as teapots and kettles are similar yet different, and whose distinct purposes my mother explained to me).

Regards,
[Intertec]<http://www.intertec.com.au/>



 Hercules Gunter - Senior Web Consultant
 Intertec Consulting - Specialists in Website Design & Web Application Development
 W: www.intertec.com.au<http://www.intertec.com.au/> • T: +61 (8) 9286 3911 • F: +61 (8) 9286 3977 • M: 0414 909 633 • A: Unit 3, 589 Stirling Hwy, Cottesloe WA 6011
 Intertec TimePro W: www.timesheets.com.au<http://www.timesheets.com.au/> • www.timesheets.co.nz<http://www.timesheets.co.nz/> • www.i-timesheets.com<http://www.i-timesheets.com/>
Working only Tuesdays to Fridays

This email and any files transmitted with it are confidential and intended solely for the use of the individual or entity to whom they are addressed

**Comment by gewarren** [comment]
Sorry for the delay. I dug up this old [documentation on differences between classes and modules](https://docs.microsoft.com/en-us/previous-versions/visualstudio/visual-studio-2008/7825002w(v=vs.90)), but I understand the bigger picture of what you're trying to convey about documentation.

**Comment by gewarren** [comment]
@gewarren The work here is to find a better home for the `project.addnewitem` F1 keyword. It should not go to this command reference topic.

**Comment by TerryGLee** [comment]
@hgunter Apologies that it's taken us some time to report back. As a colleague responded, we understand the bigger picture that you conveyed in your comments. Unfortunately, we do not have a page available that fully addresses your concerns, but we have added your request for a page that does so to our backlog to address at a future date. 

Meanwhile, we have added the `project.addnewitem` F1 keyword to another topic, [Introduction to projects and solutions](https://docs.microsoft.com/en-us/visualstudio/get-started/tutorial-projects-solutions?view=vs-2019), which includes a little more information about how to add new items to a project. Granted, this page doesn't fully address everything you've outlined, but it is a better home for it (for now) than where it was previously.

We'll close this issue now, with our thanks to you for your feedback and for helping us to improve the documentation!
